backend/worker.go

```diff
@@ -206,7 +215,17 @@ func (w *worker) processWorkItem(ctx context.Context, wi WorkItem) {
 
 	w.logger.Debugf("%v: processing work item: %s", w.Name(), wi)
 
+	if w.stop.Load() {
+		if err := w.processor.AbandonWorkItem(context.Background(), wi); err != nil {
+			w.logger.Errorf("%v: failed to abandon work item: %v", w.Name(), err)
+		}
+		return
+	}
+
 	if err := w.processor.ProcessWorkItem(ctx, wi); err != nil {
+		if w.stop.Load() {
+			ctx = context.Background()
```

Comment:
If I understand correctly, if we're fail to process a work item while we're stopping, this change will replace the current context with a new one, in which case the call to `errors.Is(err, ctx.Err())` will be false, right? Doesn't that result in an `Errorf` log during shutdown instead of a `Warnf` (which doesn't seem right)? Can you elaborate on why this specific change is being added?